### PR TITLE
HDDS-6563. Revert "HDDS-5867. Update quota when deleting open keys"

### PR DESCRIPTION
**Pull request by kaijchen**
This reverts commit 416439d7a29228db2d61e98c7f634f550f3d750c.

## What changes were proposed in this pull request?

As further discussion in HDDS-5867, if a bucket is deleted and re-created,
we should not change its quota when deleting open keys.
It's difficult for us to know whether the bucket was the original or not.

So instead of preserving quota on key creation, we decided only updating quota on key commit HDDS-6556.
Thus there is no need to update bucket quota in open key cleanup service.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6563

## How was this patch tested?

CI

### CODE REVIEW

**Comment by kaijchen** [comment]
Cc @captainzmc for review

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMOpenKeysDeleteRequest.java [hunk 1]
```diff
@@ -88,25 +87,19 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
...
-        OmBucketInfo omBucketInfo = updateTableCachePerBucket(
-            ozoneManager, trxnLogIndex, openKeyBucket, deletedOpenKeys);
-
-        if (omBucketInfo != null) {
-          String bucketKey = ozoneManager.getMetadataManager()
-              .getBucketKey(openKeyBucket.getVolumeName(),
-                  openKeyBucket.getBucketName());
-          updatedBuckets.put(bucketKey, omBucketInfo);
-        }
+        Map<String, OmKeyInfo> deleted = updateOpenKeyTableCache(ozoneManager,
```
Comment (kerneltime) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMOpenKeysDeleteRequest.java:95]: I think `deleted` can be skipped and all the keys added to `deletedOpenKeys`?

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMOpenKeysDeleteRequest.java [hunk 2]
```diff
@@ -88,25 +87,19 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
...
     Result result = null;
     Map<String, OmKeyInfo> deletedOpenKeys = new HashMap<>();
-    Map<String, OmBucketInfo> updatedBuckets = new HashMap<>();
 
     try {
       for (OpenKeyBucket openKeyBucket: submittedOpenKeyBuckets) {
         // For each bucket where keys will be deleted from,
         // get its bucket lock and update the cache accordingly.
-        OmBucketInfo omBucketInfo = updateTableCachePerBucket(
-            ozoneManager, trxnLogIndex, openKeyBucket, deletedOpenKeys);
```
Comment (kerneltime) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMOpenKeysDeleteRequest.java:98]: Why not continue to pass `deletedOpenKeys` and avoid the additional allocations?

**Review by kerneltime — APPROVED** [review]

**Review by captainzmc — APPROVED** [review]
+1 The change looks good.

**Comment by captainzmc** [comment]
Thanks @kaijchen for the patch and thanks @kerneltime for the review. Let‘s merge this.

**Comment by kaijchen** [comment]
Thanks @kerneltime and @captainzmc for the review.

**Pull request merged**